### PR TITLE
[docs] Add upgrading to 5.x guide

### PR DESCRIPTION
**Pull request by bmorelli25**
**What does this PR do?**

* Adds an upgrading guide for the APM RUM Agent version `5.x`.
* Removes documentation for the four removed config options
* Adds documentation for `eventsLimit`

**Related PRs**
* Closes https://github.com/elastic/apm-agent-rum-js/issues/671
* Related to https://github.com/elastic/apm-agent-rum-js/issues/501

**Screenshot of changes**
<details>
  <summary>Click to expand!</summary>

RUM-upgrade
</details>

### CODE REVIEW

File: docs/upgrading.asciidoc [hunk 1]
```diff
@@ -37,3 +42,75 @@ The table below is a simplified description of this policy.
...
+If you're using Elastic backend agents,
+you must upgrade them to a version that also supports the official W3C tracecontext headers.
+
+// https://github.com/elastic/apm-agent-rum-js/pull/664
+// WIP! I still need to clean this up
+Consolidated configurations::
+A single queue is now used to process all events (transactions, errors, etc.).
+This change consolidates two configuration options:
++
+* `throttleInterval` - details to come
```
Comment (vigneshshanmugam) [inline docs/upgrading.asciidoc]: Did you simplify this one?
Its 4 config in total to be precise - errorThrottleLimit, errorThrottleInterval, transactionThrottleLimit, transactionThrottleInterval.

Comment (bmorelli25) [reply docs/upgrading.asciidoc]: Oh, thanks! I missed the other two

Comment (vigneshshanmugam) [reply docs/upgrading.asciidoc]: Also we decided to rename `eventsPerMin` to `eventsLimit`.

Comment (bmorelli25) [reply docs/upgrading.asciidoc]: Awesome, thanks again for the updates 🕺

Comment (bmorelli25) [reply docs/upgrading.asciidoc]: Do we have documentation for `eventsLimit` yet?

Comment (bmorelli25) [reply docs/upgrading.asciidoc]: Added documentation for `eventsLimit` to this PR

File: docs/configuration.asciidoc [hunk 2]
```diff
@@ -212,48 +212,19 @@ var options = {
...
+[[event-throttling]]
+=== Event throttling
 
-To throttle the number of errors send to the APM Server, you can use the following options:
+Throttle the number of events sent to APM Server.
 
 [float]
-[[error-throttle-limit]]
-==== `errorThrottleLimit`
+[[event-limit]]
```
Comment (vigneshshanmugam) [inline docs/configuration.asciidoc]: We kept the config as `eventsLimit`.  Should we have to change the config?

File: docs/upgrading.asciidoc [hunk 3]
```diff
@@ -37,3 +42,90 @@ The table below is a simplified description of this policy.
...
+Removed options:
+
+* `errorThrottleLimit`
+* `errorThrottleInterval`
+* `transactionThrottleLimit`
+* `transactionThrottleInterval`
+
+Added option:
+
+* <<event-limit>> -- Configure the number of events sent sent to APM Server per minute. Defaults to `80`.
```
Comment (vigneshshanmugam) [inline docs/upgrading.asciidoc]:
```suggestion
* <<event-limit>> -- Configure the number of events sent to APM Server per minute. Defaults to `80`.
```

File: docs/upgrading.asciidoc [hunk 4]
```diff
@@ -37,3 +42,90 @@ The table below is a simplified description of this policy.
...
+
+[float]
+[[v5-breaking-changes]]
+=== 5.0 Breaking changes
+
+*Agent name changed to `rum-js`*
+
+The Agent name has been changed to `rum-js`.
+Because of this change, you may need to upgrade your APM Server instance,
+and therefor your Elasticsearch and Kibana instance.
```
Comment (jahtalab) [inline docs/upgrading.asciidoc]: Maybe mention the reason users need to upgrade their stack specially kibana. The reason is that kibana UI recognises the RUM agent name and shows RUM specific visualisation. prior to 7.x release, kibana only recognised the old name (i.e `js-base`).

File: docs/upgrading.asciidoc [hunk 5]
```diff
@@ -37,3 +42,90 @@ The table below is a simplified description of this policy.
...
+[options="header"]
+|====
+|Agent name |Agent Version
+|**Go Agent**|>= `1.6`
+|**Java Agent**|>= `1.14`
+|**.NET Agent**|>= `1.3`
+|**Node.js Agent**|>= `3.4`
+|**Python Agent**|>= `5.4`
+|**Ruby Agent**|>= `3.5`
+|====
```
Comment (jahtalab) [inline docs/upgrading.asciidoc]: Should we instead link to the compatibility table in distributed tracing, or do you think it's fine to have duplicate?

Comment (bmorelli25) [reply docs/upgrading.asciidoc]: Great question. Eventually (v6?) I envision the compatibility table in the DT guide going away. I'd like to single source the guide, but I can't until that PR is merged... OH SNAP, looks like it was. I'll rebase and make a change that single sources the compatibility table.

File: docs/upgrading.asciidoc (hunk 4 above)
Comment (bmorelli25) [reply docs/upgrading.asciidoc]: Perfect, thanks

**Comment by bmorelli25** [comment]
I've added in more breaking changes.

> However, we can merge this and add more steps in separate PRs, WDYT?

Yes, I think this makes sense.

File: docs/configuration.asciidoc (hunk 2 above)
Comment (bmorelli25) [reply docs/configuration.asciidoc]: My mistake, I missed the `s`. Fixed now.

**Pull request merged**